### PR TITLE
Add README for BinaryData

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/core/Azure.Core.Experimental/src/Primitives/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Binary Data
```
Comment (pakrym) [inline sdk/core/Azure.Core.Experimental/src/Primitives/README.md:1]: This is a strange file location.

File: sdk/core/Azure.Core.Experimental/src/Primitives/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ The ownership model of the underlying bytes varies depending on how the instance is constructed:
+ If created using the static factory method, `FromMemory(ReadOnlyMemory<byte>)`, the passed in bytes will be wrapped, rather than copied. This is useful in scenarios where performance is critical and/or ownership of the bytes is controlled completely by the consumer, thereby allowing the enforcement of whatever ownership model is needed.
+ 
+ If created using the `BinaryData(ReadOnlySpan<byte>)` constructor, `BinaryData` will maintain its own copy of the underlying bytes. This usage is geared more towards scenarios where the ownership of the bytes might be ambiguous to users of the consuming code. By making a copy of the bytes, the payload is guaranteed to be immutable. For all other constructors and static factory methods, BinaryData will assume ownership of the underlying bytes.
+
+ ## Usage
+ The main value of this type is its ability to easily convert from string to bytes to stream. This can greatly simplify API surface areas by exposing this type as opposed to numerous overloads or properties.
+ 
+ ```c#
+ var data = new BinaryData("some data");
```
Comment (pakrym) [inline sdk/core/Azure.Core.Experimental/src/Primitives/README.md:15]: Use snippets.

File: sdk/core/Azure.Core.Experimental/src/Primitives/README.md (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/core/Azure.Core.Experimental/src/Primitives/README.md:1]: I think we can move it into the Core readme eventually.

Comment (pakrym) [reply sdk/core/Azure.Core.Experimental/src/Primitives/README.md:1]: Why not put it into Azure.Core.Experimental readme and get it close to something that we would have in Azure.Core readme.

File: sdk/core/Azure.Core.Experimental/src/Primitives/README.md (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/core/Azure.Core.Experimental/src/Primitives/README.md]: Was not able to use snippet for SB code as it is in separate directory.

File: sdk/core/Azure.Core.Experimental/README.md [hunk 3]
```diff
@@ -1,3 +1,59 @@
...
+// you can also create BinaryData from a stream
+data = BinaryData.FromStream(new MemoryStream());
+
+// and convert it to a stream
+var stream = data.ToStream();
+```
+ 
+ It is also used to integrate with `ObjectSerializer`. Here is an [example](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/tests/Message/MessageLiveTests.cs#L149) from Service Bus where `BinaryData` is used to represent a message body:
+ 
+```c#
```
Comment (pakrym) [inline sdk/core/Azure.Core.Experimental/README.md:39]: We'll have to figure something out when merging to core.

**Review by pakrym — APPROVED** [review]

**Comment by tg-msft** [comment]
Can we include little to/from snippets for all the supported conversions?  I want to link people here and have them get it instantly without follow up questions.

File: sdk/core/Azure.Core.Experimental/README.md (hunk 3 above)
Comment (JoshLove-msft) [reply sdk/core/Azure.Core.Experimental/README.md]: Removed it and replaced with custom model defined in Azure.Core.Experimental test.

**Comment by JoshLove-msft** [comment]
> Can we include little to/from snippets for all the supported conversions? I want to link people here and have them get it instantly without follow up questions.

Added

**Review by tg-msft — APPROVED** [review]
Looks awesome!  Thank you.

**Pull request merged**